### DISCUSSION TITLE
Search with Regex within a specific folder

### DISCUSSION THREAD

**Discussion by joeldebruijn** [cboxdoerfer/fsearch#536] [Q&A] [answered]
I am able to search within a specific folder, just by starting the search with something like "/home/user/folderX/folderY". 

![20240204 - 042056 - Screenshot](https://github.com/cboxdoerfer/fsearch/assets/3906940/ba1da710-4abb-493b-8a7d-5a3793c4bb07)

Also searching with RegEx is succesfull, for example searching for "\d{8} - \d{6}" which matches a lot of my files using YYYYMMDD - HHMMSS naming convention.

![20240204 - 042329 - Screenshot](https://github.com/cboxdoerfer/fsearch/assets/3906940/3d26d67e-f9a0-42b4-9ef6-8b88f159b64b)

But scratching my head about a combination:

![20240204 - 042221 - Screenshot](https://github.com/cboxdoerfer/fsearch/assets/3906940/aa08e1b7-e349-456c-9502-356e1701410b)

I think I tried every combination with toggling on/off for Location and RegEx.
What am I missing?

**Comment by cboxdoerfer** [comment, marked as answer]
Hi, the problem is with the space between the path and regex query. In regex mode, spaces are treated literally and not as a logical AND. So you are basically searching for files which start with "Retail", followed by a space, then eight digits, and so on.

So what you want to do instead is:

#### If you want to search within a subfolders of `/home/user/folderX/folderY`:
1 Disable regex mode
2. Disable search in path mode
3. Use the following query: `path:/home/user/folderX/folderY regex:"\d{8} - \d{6}"`

or

#### If you want to search only within `/home/user/folderX/folderY`:
1. Keep regex mode enabled
2. Keep search in path mode enabled
3. Use the following query: `/home/user/folderX/folderY/\d{8} - \d{6}`

There are a few others ways to achieve the same goals, but that should do the trick.

**Reply by joeldebruijn** [reply]
Tested and worked, also the quotation for RegEx could have fooled me probably so thanks for the quick response!

**Comment by ajaxStardust** [comment]
Never underestimate the power of **path:**/to/desired
